dcm4che-imageio/src/main/java/org/dcm4che3/imageio/plugins/dcm/DicomMetaData.java

```diff
@@ -38,33 +38,19 @@
 
 package org.dcm4che3.imageio.plugins.dcm;
 
-import javax.imageio.metadata.IIOInvalidTreeException;
 import javax.imageio.metadata.IIOMetadata;
 
-import org.dcm4che3.data.Attributes;
 import org.w3c.dom.Node;
 
 /**
+ * Class that encapsulates the metadata for the imaging data:  this includes the DICOM attributes, as well as providing
+ * access to the pixel data.  Different strategies for loading pixel data can be supported by providing different subclasses.
+ *
  * @author Gunter Zeilinger <gunterze@gmail.com>
  *
  */
-public class DicomMetaData extends IIOMetadata {
-
-    private final Attributes fileMetaInformation;
-    private final Attributes attributes;
-
-    public DicomMetaData(Attributes fileMetaInformation, Attributes attributes) {
-        this.fileMetaInformation = fileMetaInformation;
-        this.attributes = attributes;
-    }
+public abstract class DicomMetaData extends IIOMetadata implements DicomAccessor {
```

Comment:
This is the main area of incompatibility with the current generic-config branch:  DicomMetadata is now an abstract class.  I don't think very many people are instantiating DicomMetaData directly, I believe the DicomMetaData class is generally created by DicomImageReader and then accessed from getMetaData() method, so I don't think that's a serious issue for most users.

Reply:
Is it ever used in our code base directly?  I know the goal of some is to sync generic-config up with trunk, however realistically I don't see that happening, so this might be ok.

Reply:
It turns out that the constructor WAS being used in the codebase:  the preprocessing code was creating it's own DicomMetadata object and passing it into DIcomImageReader manually.  If they were doing it, then likely other people will too.

I modified the code to restore the old constructor with the old functionality.  I should not have taken the risk of breaking that backward compatibility in the first place :(